lms/services/blackboard_api/_basic.py

```diff
@@ -83,6 +83,42 @@ def request(self, method, path):
             )
             return self._send(method, url)
 
+    def async_requests(self, method, urls):
+        import asyncio
+
+        import aiohttp
+
+        headers = {}
+        access_token = self._oauth_http_service._oauth2_token_service.get().access_token
+        headers["Authorization"] = f"Bearer {access_token}"
+
+        async def async_request(aio_session, method, url, headers=None):
+            try:
+                async with aio_session.request(
+                    method, url, headers=headers
+                ) as response:
+                    return response
+            except aiohttp.ClientError:
+                raise ExternalRequestError("Blackboard async request failed")
```

Comment:
This won't have as many details as the rest of `ExternalRequestError` (request and responses objects are different form the ones from `requests` so it's not trivial to attach them to the exception) but I hope the fact that we only use this in one place ATM and stack trace in celery would provide enough details.

Reply:
I think you probably want an explicit `raise ExternalRequestError(...) from err)`